src/inspect_ai/model/_providers/bedrock.py

```diff
@@ -9,28 +8,54 @@
     DEFAULT_MAX_TOKENS,
     DEFAULT_TIMEOUT,
 )
+from inspect_ai._util.content import Content, ContentImage, ContentText
 from inspect_ai._util.error import pip_dependency_error
+from inspect_ai._util.images import image_as_data
 from inspect_ai._util.version import verify_required_version
+from inspect_ai.model._providers.util.converse_model import (
+    ClientConverseRequest,
+    Image,
+    ImageFormat,
+    ImageSource,
+    InferenceConfig,
+    Message,
+    MessageContent,
+    Response,
+    StopReason,
+    SystemContent,
+    Tool,
+    ToolConfig,
+    ToolResult,
+    ToolResultContent,
+    ToolSpec,
+    ToolUse,
+)
+from inspect_ai.model._providers.util.converse_model import (
+    ToolChoice as ConverseToolChoice,
+)
 from inspect_ai.tool import ToolChoice, ToolInfo
+from inspect_ai.tool._tool_call import ToolCall
+from inspect_ai.tool._tool_choice import ToolFunction
 
 from .._chat_message import (
     ChatMessage,
+    ChatMessageAssistant,
+    ChatMessageSystem,
+    ChatMessageTool,
+    ChatMessageUser,
 )
 from .._generate_config import GenerateConfig
 from .._model import ModelAPI
 from .._model_call import ModelCall
-from .._model_output import ChatCompletionChoice, ModelOutput, ModelUsage
+from .._model_output import (
+    ChatCompletionChoice,
+    ModelOutput,
+    ModelUsage,
+)
 from .util import (
-    ChatAPIHandler,
-    ChatAPIMessage,
-    Llama31Handler,
-    as_stop_reason,
-    chat_api_input,
     model_base_url,
 )
 
-ANTHROPIC_API_KEY = "ANTHROPIC_API_KEY"
-
 
 class BedrockAPI(ModelAPI):
```

Comment:
Where do `model_args` go for Bedrock? Feels like they could go to `aioboto3.Session()` or more likely go to `self.session.Client`?

Reply:
No, you shouldn't need that at all for Bedrock as it authenticates natively w/ AWS credentials (previously we had that in there when using the Anthropic client)